### PR TITLE
[W-11392825] - sa11y testing for NPSP - Donation History + Giving Summary

### PR DESCRIPTION
**Pull request by JoseRossiSF**
# Technical Document
[Technical Design Review - W-11392825 - NPSP](https://salesforce.quip.com/K8v2ARbvjIOb)
Adding sa11y jest test for components and ensure the tests are passing and the accessibility of the components is standards compliant.

# Changes

# Work Item
[W-11392825](https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000010oQGDYA2/view)

### CODE REVIEW

File: force-app/main/default/lwc/donationHistoryTable/__tests__/donationHistoryTable.test.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
 import { createElement } from 'lwc';
 import DonationHistoryTable from '../donationHistoryTable';
+import { extended, full } from "@sa11y/preset-rules";
+import { setup } from "@sa11y/jest";
 const MOCK_DATA = require('./data/donationHistoryTableData.json');
 
+beforeAll(() => {
```
Comment (andrewyu-salesforce) [inline force-app/main/default/lwc/donationHistoryTable/__tests__/donationHistoryTable.test.js:7]: This function is causing the existing Sa11y test suite to fail with the following error message.
`Cannot add a hook after tests have started running. Hooks must be defined synchronously.`

**Review by andrewyu-salesforce — COMMENTED** [review]
The changed

File: force-app/main/default/lwc/donationHistoryTable/__tests__/donationHistoryTable.test.js (hunk 1 above)
Comment (JoseRossiSF) [reply force-app/main/default/lwc/donationHistoryTable/__tests__/donationHistoryTable.test.js:7]: Hi Andrew. Thanks for the suggestion. This has now been fixed. Please, re-review.

Comment (lparrott) [reply force-app/main/default/lwc/donationHistoryTable/__tests__/donationHistoryTable.test.js:7]: I'm still seeing the same issue in this test suite: `Cannot add a hook after tests have started running. Hooks must be defined synchronously.`

Comment (JoseRossiSF) [reply force-app/main/default/lwc/donationHistoryTable/__tests__/donationHistoryTable.test.js]: @lparrott You are right. I had solved the issue for Giving Summary component, but not for Donation History table component. It should be working now

**Review by andrewyu-salesforce — COMMENTED** [review]
It seem like `givingSummary` component is not passing sa11y accessibility test. Can you also address that?

**Comment by JoseRossiSF** [comment]
> It seem like `givingSummary` component is not passing sa11y accessibility test. Can you also address that?

@andrewyu-salesforce Thank you, Andrew. Since sa11y was passing locally in Jest, I had not noticed the github action was failing. I have fixed the aria-role-parenting issue and can see the tests for both Donation History and Giving Summary are passing.

Please let me know if you notice anything else :)

**Review by andrewyu-salesforce — APPROVED** [review]
looks good

**Pull request merged**